backends/ebpf/runtime/pcap_util.c

```diff
@@ -239,3 +240,18 @@ void sort_pcap_list(pcap_list_t *pkt_list) {
     /* Sort the master list and return the meta information*/
     qsort(pkt_list->pkts, pkt_list->len, sizeof(pcap_pkt *), compare_pkt_time);
 }
+
+char *generate_pcap_name(const char *pcap_base, int index, const char *suffix) {
+    int file_name_max  = 256;   // Max filename length
+    int max_10_uint32  = 10;     // Max size of uint32 in base 10
+    /* Dynamic string length plus max decimal representation of uint16_t */
+    int file_length = strlen(pcap_base) + strlen(suffix) + max_10_uint32 + 1;
+    char *pcap_name = malloc(file_length);
+    int offset = snprintf(pcap_name, file_length,"%s%hu%s",
+                    pcap_base, index, suffix);
+    if (offset >= file_name_max) {
```

Comment:
This is wrong, you should compare offset with file_length not with file_name_max.
If offset is between the two values you will not report an error.
In fact, file_name_max is useless, you should remove it.